app/views/patients/pledge.js.erb

```diff
@@ -0,0 +1,16 @@
+$('.modal-content').html('')
+$('.modal').attr('id', 'pledge-modal')
+$('.modal-content').append(
+  '<%= escape_javascript(render partial: "patients/pledge", locals: { patient: @patient, disable_next: disable_continue?(@patient) })%>'
+);
+
+$('.modal').modal('toggle');
```

Comment:
Got it. I tend to go 50-50 on clearly named functions for stuff like this, just because JS is so blasted hard to read, but I hear you on the specificity of this. Let's merge as is.